setup.py

```diff
@@ -1,19 +1,164 @@
 import os
 import sys
+import platform
+import subprocess
+import git
+
+from distutils.core import setup, Extension 
 
 if os.name == 'nt':
     from distutils import msvc9compiler
     msvc9compiler.VERSION = 14.0 #Visual studio 2015
 
-if sys.version > '3':
-    setup_file = "setup_3.py"
+if sys.version <= '2':
+    print('No more supports python2, use python3 instead')
+    sys.exit(1)
+
+# Get the script directory.
+def get_script_dir():
+    path = os.path.abspath(sys.argv[0])
+    if os.path.isdir(path):
+        return path
+    elif os.path.isfile(path):
+        return os.path.dirname(path)
+
+# Update
+if os.path.isfile("cci-src/configure") == 0:
+    if sys.version_info >= (3, 7):
```

Comment:
Should the submodule update method be used differently depending on the python version?
I do not know, so please inquire.

Reply:
GitPython submodule's minimum required version is 3.7
so if Python version is less than 3.7, using subprocess command for update